ydb/core/base/pool_stats_collector.cpp

```diff
@@ -51,14 +51,27 @@ class TStatsCollectingActor : public NActors::TStatsCollectingActor {
     };
 
     void OnWakeup(const TActorContext &ctx) override {
+        static auto whiteboardId = NNodeWhiteboard::MakeNodeWhiteboardServiceId(ctx.SelfID.NodeId());
         MiniKQLPoolStats.Update();
 
-        TVector<std::tuple<TString, double, ui32, ui32>> pools;
+        auto systemUpdate = std::make_unique<NNodeWhiteboard::TEvWhiteboard::TEvSystemStateUpdate>();
+        ui32 coresTotal = 0;
+        double coresUsed = 0;
         for (const auto& pool : PoolCounters) {
-            pools.emplace_back(pool.Name, pool.Usage, pool.Threads, pool.LimitThreads);
+            auto& pb = *systemUpdate->Record.AddPoolStats();
+            pb.SetName(pool.Name);
+            pb.SetUsage(pool.Usage);
+            pb.SetThreads(static_cast<ui32>(pool.Threads));
+            pb.SetLimit(static_cast<ui32>(pool.LimitThreads));
+            if (pool.Name != "IO") {
+                coresTotal += static_cast<ui32>(pool.DefaultThreads);
+            }
+            coresUsed += pool.Usage * pool.LimitThreads;
```

Comment:
coresUsed can exceed coresTotal? :)

Reply:
> coresUsed can exceed coresTotal? :)

technically - yes, they can. for example, when we have events taking longer than 1 second to process